src/DiffSharp.Core/Util.fs

```diff
@@ -121,7 +278,7 @@ let inline arraysAllClose (relativeTolerance:'T) (absoluteTolerance:'T) (array1:
     let dim1 = array1.Length
     let dim2 = array2.Length
     if dim1 <> dim2 then false
-    else Array.map2 (fun a b -> abs(a-b) <= absoluteTolerance + relativeTolerance*abs(b)) array1 array2 |> Array.forall id
+    else Array.map2 (fun a b -> abs(a-b) < absoluteTolerance + relativeTolerance*abs(b)) array1 array2 |> Array.forall id
```

Comment:
The `<=` that I see replaced in this commit with `<` ultimately comes from https://numpy.org/doc/1.18/reference/generated/numpy.isclose.html and https://docs.python.org/dev/library/math.html#math.isclose that PyTorch also inherits.

It would be good to keep it as `<=`